shared/lib/transactions-controller-utils.js

```diff
@@ -135,3 +138,49 @@ export const TRANSACTION_ENVELOPE_TYPE_NAMES = {
   FEE_MARKET: 'fee-market',
   LEGACY: 'legacy',
 };
+
+let registry;
+
+/**
+ * Attempts to return the method data from the MethodRegistry library, the message registry library and the token abi, in that order of preference
+ *
+ * @param {string} fourBytePrefix - The prefix from the method code associated with the data
+ * @param {boolean} allow4ByteRequests - Whether or not to allow 4byte.directory requests, toggled by the user in privacy settings
+ * @param {object} provider - Provider for current network
+ * @returns {object}
+ */
+export async function getMethodDataAsync(
```

Comment:
Rather than moving this to a legacy JS file, would this be better placed in `shared/lib/four-byte.ts` so we can include the type also?

Reply:
PR is updated